### PR TITLE
Circle build npm publishing

### PR DESCRIPTION
**Pull request by elliottcrush**
#### What does this PR do? (please provide any background)
Adds circle scripts to automatically publish brands to npm when the version number is bumped. This also merges in some more changes to `ui-brand-hx` that were recently merged.

#### What tests does this PR have?
None

#### How can this be tested?
Checkout the branch and change the deployment branch in `circle.yml` to point to the `circle` branch. Then changes a version number in one of the brands `package.json` and push the branch. This is the only way to fully test that the new versions have been published correctly.

#### Screenshots / Screencast
screen shot 2016-01-26 at 17 18 35

#### What gif best describes how you feel about this work?
giphy

---

- [x] I have checked our general [contributing document](https://github.com/holidayextras/culture/blob/master/CONTRIBUTING.md) and the project specific [contributing document](../blob/master/CONTRIBUTING.md) (if present) and I'm happy for this to be reviewed.

#### Reviewers

**Review 1**
- [x] :+1:

**Review 2** \*
- [x] :+1:

**Review 3** _(optional)_
- [ ] :+1:

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.

\*  for HX this review must be completed by an SE, SA or Project Guru

### CODE REVIEW

**Comment by Nick-Patrick** [comment]
:+1:  I'm happy with this

File: brands/default/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@holidayextras/default-brand",
-  "version": "1.0.0",
+  "version": "1.0.3",
```
Comment (elliottcrush) [inline brands/default/package.json]: Just though i'd put a comment in here before it's raised. This is due to the fact that we were using this default brand to test the NPM publishing locally, that is why its gone from 1.0.0 to 0.3

**Comment by shackpank** [comment]
:two:

File: README.md [hunk 2]
```diff
@@ -14,4 +16,14 @@
...
\ No newline at end of file
+```
+### Publishing brand changes to NPM
+
+All the brands within this repository are public NPM modules. See [here for an example](https://www.npmjs.com/package/@holidayextras/hx-brand).
+
+In order to publish your local changes to NPM you'll need to do the following:
+
+1. Make the changes to the brand less as you would normally do
+2. Edit the version number in the `package.json` file of the brand you've just created. Please refer to the culture repo for more information on how to version correctly.
```
Comment (shackpank) [inline README.md]: Is there a specific document/section this could link to?

Comment (elliottcrush) [inline README.md]: Potentially - https://docs.npmjs.com/getting-started/semantic-versioning ? It looks we point our culture docs that way currently.

**Comment by shackpank** [comment]
Do any of these builds show the publish succeeding? (the last one doesn't attempt it)

There's some style/lint-type inconsistency around `key:value` and `key: value`, not a blocker, just picked up on it - might be nice to make consistent now if we're looking to lint in the near future to make that introduction a little easier.

File: README.md (hunk 2 above)
Comment (shackpank) [inline README.md]: That could work, or [the paragraph here linking it](https://github.com/holidayextras/culture/blob/master/publishing-to-npm.md#updating-actively-used-modules), although you'd have to skip over some of it. Separating out the general guidelines from the process for manually releasing new versions could help when we're looking for something to link to in a repo that has some publish automation setup.

Comment (elliottcrush) [inline README.md]: I agree, I will link to the [this](https://github.com/holidayextras/culture/blob/master/publishing-to-npm.md#updating-actively-used-modules) for the time being and put a PR in to split the section into automation setup / guidelines for releasing.

**Comment by elliottcrush** [comment]
The npm publishing only works upon merge to master, it won't attempt it for this branch. There is no notification of this actually publishing however, any thoughts on exactly how we'd do that?

Where specifically are you spotting the key:value inconsistencies? Happy to fix now.

File: brands/holidayextras/src/less/_navbar.less [hunk 3]
```diff
@@ -23,29 +23,56 @@
     overflow: hidden;
     text-overflow: ellipsis;
   }
+  a {
+    text-decoration:none;
```
Comment (shackpank) [inline brands/holidayextras/src/less/_navbar.less]: This file, and `_dropdown.less` use
```
text-decoration:none
              -><-       
```

The other files in the PR use:
```
text-decoration: none;
              -> <-
```

**Comment by shackpank** [comment]
What sort of notifications do you mean? I was more after an example to see what the build log output looked like, things like posting to slack channels can definitely wait.

Have commented inline on the "linting"

**Comment by elliottcrush** [comment]
Ah sorry. I misunderstood you - you can see the build log for the publishing test I did on this branch yesterday [here](https://circleci.com/gh/holidayextras/brands/15).

Log output example:
<code>
sh ./scripts/circle.sh
publish info Local version: 1.0.0 
publish info Published version: 1.0.0 
publish info Your local version is the same as your published version: publish will do nothing 
publish info Local version: 1.0.1 
publish info Published version: 1.0.1 
publish info Your local version is the same as your published version: publish will do nothing 
publish info Local version: 1.0.3 
publish info Published version: 1.0.2 
+ @holidayextras/default-brand@1.0.3
publish info published ok 
publish info Local version: 1.0.0 
publish info Published version: 1.0.0 
publish info Your local version is the same as your published version: publish will do nothing 
publish info Local version: 1.0.0 
publish info Published version: 1.0.0 
publish info Your local version is the same as your published version: publish will do nothing 
publish info Local version: 1.0.4 
publish info Published version: 1.0.4 
publish info Your local version is the same as your published version: publish will do nothing 
</code>

**Comment by elliottcrush** [comment]
Have made some more changes to make `key: value` consistent @shackpank

**Comment by shackpank** [comment]
:+1:

**Pull request merged**